lib/cli/ui/widgets/status.rb

```diff
@@ -0,0 +1,55 @@
+# frozen-string-literal: true
+require('cli/ui')
+
+module CLI
+  module UI
+    module Widgets
+      class Status < Widgets::Base
+        ARGPARSE_PATTERN = %r{
+          \A (?<succeeded> \d+)
+          :  (?<failed>    \d+)
+          :  (?<working>   \d+)
+          :  (?<pending>   \d+) \z
+        }x # e.g. "1:23:3:404"
+
+        OPEN  = Color::RESET.code + Color::BOLD.code + '[' + Color::RESET.code
+        CLOSE = Color::RESET.code + Color::BOLD.code + ']' + Color::RESET.code
+
+        SPINNER_STOPPED = CLI::UI::Spinner::RUNES[4]
+
+        def render
+          #   [          0✓               2✗             3⠼              4⌛︎           ]
+          "#{OPEN}#{succeeded_part} #{failed_part} #{working_part} #{pending_part}#{CLOSE}"
+        end
+
+        private
+
+        def zero?(num_str)
+          num_str == '0'
+        end
+
+        def colorize_if_nonzero(num_str, glyph, color)
+          color = Color::GRAY if zero?(num_str)
+          color.code + num_str + glyph
+        end
+
+        def succeeded_part
+          colorize_if_nonzero(@succeeded, Glyph::CHECK.char, Color::GREEN)
+        end
+
+        def failed_part
+          colorize_if_nonzero(@failed, Glyph::X.char, Color::RED)
+        end
+
+        def working_part
+          rune = zero?(@working) ? SPINNER_STOPPED : Spinner.current_rune
```

Comment:
🚲 🏠: Why do extra work to force it to a particular state when stopped?  I'd be more inclined to just let it be whatever it stops on.

Reply:
The way I think about it is that the spinner indicates an active operation, and when there's no active operation in that basket, it makes more sense to me that the spinner should "pause". I felt like it made a bit more sense visually but I'm not super attached to it.

Reply:
Even more 🚲🏠, could we change the `SPINNER_STOPPED` to something like ⠿ instead? Using a rune that is normally used to indicate "working" makes it looks like it froze.

Reply:
I wasn't suggesting keeping it spinning, just stopping wherever it happened to land, but I like @Wai-Lau's suggestion better.